lib/src/editorhelper.cpp

```diff
@@ -131,92 +146,207 @@ void resize_buffer(T* src_buffer, T* dst_buffer, int src_w, int src_h, int dst_w
+
+template void rotate_buffer_clockwise_90<uint8_t>(uint8_t*, uint8_t*, int, int, int, int, int, int);
+template void rotate_buffer_clockwise_90<uint16_t>(uint16_t*, uint16_t*, int, int, int, int, int,
+                                                   int);
+template void rotate_buffer_clockwise_90<uint32_t>(uint32_t*, uint32_t*, int, int, int, int, int,
+                                                   int);
+template void rotate_buffer_clockwise_90<uint64_t>(uint64_t*, uint64_t*, int, int, int, int, int,
+                                                   int);
+template void rotate_buffer_clockwise_180<uint8_t>(uint8_t*, uint8_t*, int, int, int, int, int,
+                                                   int);
+template void rotate_buffer_clockwise_180<uint16_t>(uint16_t*, uint16_t*, int, int, int, int, int,
+                                                    int);
+template void rotate_buffer_clockwise_180<uint32_t>(uint32_t*, uint32_t*, int, int, int, int, int,
+                                                    int);
+template void rotate_buffer_clockwise_180<uint64_t>(uint64_t*, uint64_t*, int, int, int, int, int,
+                                                    int);
+template void rotate_buffer_clockwise_270<uint8_t>(uint8_t*, uint8_t*, int, int, int, int, int,
+                                                   int);
+template void rotate_buffer_clockwise_270<uint16_t>(uint16_t*, uint16_t*, int, int, int, int, int,
+                                                    int);
+template void rotate_buffer_clockwise_270<uint32_t>(uint32_t*, uint32_t*, int, int, int, int, int,
+                                                    int);
+template void rotate_buffer_clockwise_270<uint64_t>(uint64_t*, uint64_t*, int, int, int, int, int,
+                                                    int);
+template void resize_buffer<uint8_t>(uint8_t*, uint8_t*, int, int, int, int, int, int);
+template void resize_buffer<uint16_t>(uint16_t*, uint16_t*, int, int, int, int, int, int);
+template void resize_buffer<uint32_t>(uint32_t*, uint32_t*, int, int, int, int, int, int);
+template void resize_buffer<uint64_t>(uint64_t*, uint64_t*, int, int, int, int, int, int);
+
+uhdr_mirror_effect::uhdr_mirror_effect(uhdr_mirror_direction_t direction) : m_direction{direction} {
+  m_mirror_horizontal_uint8_t = mirror_buffer_horizontal<uint8_t>;
+  m_mirror_horizontal_uint16_t = mirror_buffer_horizontal<uint16_t>;
+  m_mirror_horizontal_uint32_t = mirror_buffer_horizontal<uint32_t>;
+  m_mirror_horizontal_uint64_t = mirror_buffer_horizontal<uint64_t>;
+
+  m_mirror_vertical_uint8_t = mirror_buffer_vertical<uint8_t>;
+  m_mirror_vertical_uint16_t = mirror_buffer_vertical<uint16_t>;
+  m_mirror_vertical_uint32_t = mirror_buffer_vertical<uint32_t>;
+  m_mirror_vertical_uint64_t = mirror_buffer_vertical<uint64_t>;
+}
+
+uhdr_rotate_effect::uhdr_rotate_effect(int degree) : m_degree{degree} {
+  m_rotate_90_uint8_t = rotate_buffer_clockwise_90<uint8_t>;
```

Comment:
I prefer making the degree a separate input parameter instead of a separate method. Reason is that we may support other degrees that are not multiple of 90 in the future after we bring OpenGL into the library.

Reply:
Done